spikeinterface/comparison/basemulticomparison.py

```diff
@@ -0,0 +1,185 @@
+from typing import OrderedDict
+import numpy as np
+import networkx as nx
+from copy import deepcopy
+from .basecomparison import BaseComparison
+
+
+class BaseMultiComparison(BaseComparison):
+    """
+    Compares multiple comparison objects using a graph approach.
+
+    - Pair-wise comparisons are made
+    - An agreement graph is built based on the agreement score
+
+    Parameters
+    ----------
+    object_list: list
+        List of objects to be compared
+    name_list: list
+        List of object names
+    """
+
+    def __init__(self, object_list, name_list=None):
+        self._object_list = object_list
+        self.name_list = name_list
+        self.graph = None
+        self.subgraphs = None
+        self.clean_graph = None
+
+    def _compute_all(self):
+        self._do_comparison()
+        self._do_graph()
+        self._clean_graph()
+        self._do_agreement()
+
+    def _compare_ij(self, i, j):
+        raise NotImplementedError
+
+    def _populate_nodes(self):
+        raise NotImplementedError
+
+    @property
+    def units(self):
+        return deepcopy(self._new_units)
+
+    def compute_subgraphs(self):
+        """
+        Computes subgraphs of connected components.
+
+        Returns
+        -------
+        sg_object_names: list
+            List of sorter names for each node in the connected component subgraph
+        sg_units: list
+            List of unit ids for each node in the connected component subgraph
+        """
+        if self.clean_graph is not None:
+            g = self.clean_graph
+        else:
+            g = self.graph
+        subgraphs = (g.subgraph(c).copy() for c in nx.connected_components(g))
+        sg_object_names = []
+        sg_units = []
+        for sg in subgraphs:
+            object_names = []
+            unit_names = []
+            for node in sg.nodes:
+                object_names.append(node[0])
+                unit_names.append(node[1])
+            sg_object_names.append(object_names)
+            sg_units.append(unit_names)
+        return sg_object_names, sg_units
+
+    def _do_comparison(self, ):
+        # do pairwise matching
+        if self._verbose:
+            print('Multicomaprison step 1: pairwise comparison')
+
+        self.comparisons = []
+        for i in range(len(self._object_list)):
+            for j in range(i + 1, len(self._object_list)):
+                if self._verbose:
+                    if self.name_list is not None:
+                        name_i = self.name_list[i]
+                        name_j = self.name_list[j]
```

Comment:
I think we should aways have a name and this should be handle in BaseComparison.